bugbug/models/rcatype.py

```diff
@@ -0,0 +1,167 @@
+            bug_features.is_coverity_issue(),
+            bug_features.has_crash_signature(),
+            bug_features.has_url(),
+            bug_features.has_w3c_url(),
+            bug_features.has_github_url(),
+            bug_features.whiteboard(),
+            bug_features.patches(),
+            bug_features.landings(),
+            bug_features.blocked_bugs_number(),
+            bug_features.ever_affected(),
+            bug_features.affected_then_unaffected(),
+            bug_features.product(),
+            bug_features.component(),
+        ]
+
+        cleanup_functions = [
+            feature_cleanup.url(),
+            feature_cleanup.fileref(),
+            feature_cleanup.synonyms(),
+        ]
+
+        self.extraction_pipeline = Pipeline(
+            [
+                (
+                    "bug_extractor",
+                    bug_features.BugExtractor(feature_extractors, cleanup_functions),
+                ),
+                (
+                    "union",
+                    ColumnTransformer(
+                        [
+                            ("data", DictVectorizer(), "data"),
+                            ("title", self.text_vectorizer(min_df=0.001), "title"),
+                            (
+                                "first_comment",
+                                self.text_vectorizer(min_df=0.001),
+                                "first_comment",
+                            ),
+                            (
+                                "comments",
+                                self.text_vectorizer(min_df=0.001),
+                                "comments",
+                            ),
+                        ]
+                    ),
+                ),
+            ]
+        )
+
+        self.clf = OneVsRestClassifier(xgboost.XGBClassifier(n_jobs=16))
+
+    def get_labels(self):
+        classes = {}
+
+        for bug_data in bugzilla.get_bugs():
+            target = np.zeros(len(self.RCA_LIST))
+            whiteboard_data = re.sub(" ", "", bug_data["whiteboard"]).lower()
+            for whiteboard in whiteboard_data.split("["):
+                if whiteboard.startswith("rca-"):
+                    rca_whiteboard = re.sub("]", "", whiteboard)
+                    if rca_whiteboard.split("-", 1)[1] not in self.RCA_LIST:
+                        logger.warning(
+                            rca_whiteboard.split("-", 1)[1] + " not in RCA_LIST"
+                        )
+                    else:
+                        target[self.RCA_LIST.index(rca_whiteboard.split("-", 1)[1])] = 1
+
+                elif whiteboard.startswith("rca:"):
+                    if re.sub("]", "", whiteboard).split(":")[1] in self.RCA_LIST:
+                        target[
+                            self.RCA_LIST.index(
+                                re.sub("]", "", whiteboard).split(":")[1]
+                            )
+                        ] = 1
+
```

Comment:
Print a warning if `rca` is in whiteboard but we were not able to find a class for it